app/services/user_validation_service.rb

```diff
@@ -46,12 +46,14 @@ def validate_user(user, task_id = nil, request = nil, authenticate_options = {})
 
     return validate_user_handle_not_ready(db_user) unless server_ready?
 
+    startpage = start_url_for_user(start_url)
+
     # Start super admin at the main db if the main db has no records yet
     return validate_user_handle_no_records if db_user.super_admin_user? &&
                                               ::Settings.product.maindb &&
-                                              !::Settings.product.maindb.constantize.first
+                                              !::Settings.product.maindb.constantize.first &&
+                                              !startpage
```

Comment:
[cosmetic] I find the resulting flow hard to read.  If startpage is non-nil, 2 returns will be skipped and only then successful result is returned.  How about early successful return right here:
```
return ValidateResult.new(:pass, nil, startpage) if startpage
```

Ignore me if someone who knows this code thinks differently :-)